lib/services/common/securityServiceOAuth2.js

```diff
@@ -64,19 +66,29 @@ function auth(trust, callback) {
     };
 
     logger.debug(context, 'Authentication on the OAuth2 provider [%s] using trust [%s]', url, trust);
+    logger.debug(context, 'Using the following request:\n\n%s\n\n', JSON.stringify(options, null, 4));
 
     request(options, function handleResponse(error, response, body) {
         if (error) {
             logger.error(context, 'OAUTH2-001: Error retrieving token from OAuth2 provider: %s', error);
             callback(new errors.TokenRetrievalError(trust, error));
-        } else if (response.statusCode === 201) {
+        } else if (response.statusCode === 201 || response.statusCode === 200 /* Keyrock response */) {
```

Comment:
in this case it looks that the keycloak and auth are using wrong code:
https://tools.ietf.org/html/rfc6749#section-5.1

Reply:
OK. NTC